devtools/platform-descriptor-json-plugin/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java

```diff
@@ -121,29 +112,25 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             return;
         }
 
-        // Read the overrides file for the extensions (if it exists)
-        Map<String, JsonObject> extOverrides = new HashMap<>();
-        JsonObject theRest = null;
-        if (overridesFile.isFile()) {
-            info("Found overrides file %s", overridesFile);
-            try (JsonReader jsonReader = Json.createReader(new FileInputStream(overridesFile))) {
-                JsonObject overridesObject = jsonReader.readObject();
-                JsonArray extOverrideObjects = overridesObject.getJsonArray("extensions");
-                if (extOverrideObjects != null) {
-                    // Put the extension overrides into a map keyed to their GAV
-                    for (JsonValue val : extOverrideObjects) {
-                        JsonObject extOverrideObject = val.asJsonObject();
-                        String key = extensionId(extOverrideObject);
-                        extOverrides.put(key, extOverrideObject);
-                    }
-                }
+        List<File> overrides = new ArrayList<>();
 
-                theRest = overridesObject;
-            } catch (IOException e) {
-                throw new MojoExecutionException("Failed to read " + overridesFile, e);
+        overrides.addAll(Arrays.stream(overridesFile.split(","))
+                .map(s -> {
+                    return new File(s.trim());
+                })
+                .collect(Collectors.toList()));
+
+        List<OverrideInfo> allOverrides = new ArrayList();
+        for (File file : overrides) {
```

Comment:
You can iterate `overridesFile.split(",")` and delete the `List<File> overrides` above :)

Reply:
true - good catch. it became this way as I was originally merging two lists but that’s not relevant anymore fixing.